sci_analysis/graphs/vector.py

```diff
@@ -262,19 +262,31 @@ def __init__(self, xdata, ydata=None, **kwargs):
         :param _points: Display the scatter points.
         :param _contours: Display the density contours
         :param _boxplot_borders: Display the boxplot borders
+        :param _highlight: an array-like with points to highlight based on labels
+        :param _labels: a vector object with the graph labels
         :param _title: The title of the graph.
         :param _save_to: Save the graph to the specified path.
         :return: pass
         """
         self._fit = kwargs['fit'] if 'fit' in kwargs else True
         self._points = kwargs['points'] if 'points' in kwargs else True
+        self._labels = kwargs['labels'] if 'labels' in kwargs else [False]
+        self._highlight = kwargs['highlight'] if 'highlight' in kwargs else None
         self._contours = kwargs['contours'] if 'contours' in kwargs else False
         self._contour_props = (31, 1.1)
         self._boxplot_borders = kwargs['boxplot_borders'] if 'boxplot_borders' in kwargs else False
         self._title = kwargs['title'] if 'title' in kwargs else 'Bivariate'
         self._save_to = kwargs['save_to'] if 'save_to' in kwargs else None
         yname = kwargs['yname'] if 'yname' in kwargs else 'y Data'
         xname = kwargs['xname'] if 'xname' in kwargs else 'x Data'
+        if 'labels' in kwargs:
+            if len(kwargs['labels']) != len(xdata):
+                raise AttributeError('The length of passed labels does not match the length of xdata and ydata')
+            if ndarray.any(kwargs['labels'].index != xdata.data.index):
```

Comment:
If drop_nan(), which runs as part of the Vector object constructor, removes values from the internal pandas DataFrame, the index and length might not match kwargs['labels']. One thought is to join labels on index to the DataFrame which will remove any index values that are not in the DataFrame. You would definitely need to write a  unit test to make sure it works correctly.

Reply:
I'm trying to work around this by adding a "dropped_vals" attribute to the numeric constructor, so when it runs drop_nan, it also creates a mask of what was dropped. Then, we can remove those values from the labels column, whether the labels are a pandas series, or a numpy array. 

It seems to work fine when I test it locally, but unit tests don't seem to recall the "dropped_vals"attribute  when it comes to run graphgroupscatter, they say the labels array doesn't have that attribute.